src/ui/Facet/Facet.ts

```diff
@@ -1528,6 +1528,35 @@ export class Facet extends Component {
     this.dependsOnManager = new DependsOnManager(facetInfo);
   }
 
+  private dependsOnUpdateParentDisplayValue() {
+    if (this.options.dependsOn) {
+      const masterFacetHTML = $$(this.root)
+        .findAll(`.CoveoFacet`)
+        .filter(HTMLFacet => {
+          const facet = <Facet>Facet.get(HTMLFacet);
+          return facet.options.id === this.options.dependsOn;
+        });
+      if (!masterFacetHTML.length) {
+        this.logger.warn(
+          `Unable to find a Facet with the id or field "${this.options.dependsOn}".`,
+          `The master Facet values can't be updated.`
+        );
+        return;
+      }
+      if (masterFacetHTML.length > 1) {
```

Comment:
Would it be better to catch this error of two facets having the same id/field on initialisation rather than if the `dependsOn` option is configured?

In terms of the message, could logging the facet titles could be helpful for someone trying to resolve the warning? e.g.
```
The following facets have the same id/field: facetA, facetB, facetC.
A facet can only depend on one other facet. Please ensure the above facets have ids that are unique.
The master facet values can't be updated.
```
```

Reply:
🤔 🤔  trying to do this and have a question. Where should I do the check because I tried in the init but it explodes because the other facets are not yet initialized?

Reply:
Hmm, I was thinking of using the `afterComponentsInitialization` event. Another option might be `afterInitialization`. There's a nice [documentation page here](https://docs.coveo.com/en/417/javascript-search-framework/javascript-search-framework-events) explaining the order events.

Reply:
ok, I'll check those out. thanks